polaris-core/src/main/java/org/apache/polaris/core/config/FeatureConfiguration.java

```diff
@@ -209,6 +209,20 @@ public static void enforceFeatureEnabledOrThrow(
           .defaultValue(true)
           .buildFeatureConfiguration();
 
+  public static final FeatureConfiguration<Boolean> ENABLE_XTABLE_REST_SERVICE =
+      PolarisConfiguration.<Boolean>builder()
+          .key("ENABLE_XTABLE_REST_SERVICE")
+          .description(
+              "If true, delegates table metadata conversion to an external XTable REST Service")
```

Comment:
Why would the Polaris catalog service have a feature flag that controls whether or not the XTable service is "enabled" (running)?

We _do_ need feature flags to control a few behaviors:

1. Convert-on-write
2. Convert-on-read

Reply:
However we implement this, it should probably live in application.properties instead of being a feature configuration checked at runtime.

It also needs to allow different conversion implementations to be specified. So something like:

```
table-conversion.type = xtable
table-conversion.host=www.my-xtable.com
table.conversion.supported-types=delta,hudi,iceberg
```

or:

```
table-conversion.type=local
table.conversion.supported-types=delta,iceberg
```

Check out how the `RealmContextResolver` is configured for example.

Reply:
> Why would the Polaris catalog service have a feature flag that controls whether or not the XTable service is "enabled" (running)?
> 
> We _do_ need feature flags to control a few behaviors:
> 
> 1. Convert-on-write
> 2. Convert-on-read

I think the naming might be off i think what you wrote for TABLE_CONVERSION_TYPE=XTable makes sense to me.

Can you help me understand in Polaris what is the distinction between application properties vs a feature configuration?